packages/editor/src/components/rich-text/format-edit.js

```diff
@@ -0,0 +1,103 @@
+/**
+ * WordPress dependencies
+ */
+import { Component, Fragment } from '@wordpress/element';
+import { getActiveFormat, getFormatTypes } from '@wordpress/rich-text';
+import { Fill, KeyboardShortcuts, ToolbarButton } from '@wordpress/components';
+import { rawShortcut } from '@wordpress/keycodes';
+
+/**
+ * Internal dependencies
+ */
+import InserterListItem from '../inserter-list-item';
+import { normalizeTerm } from '../inserter/menu';
+
+function isResult( { title, keywords = [] }, filterValue ) {
+	const normalizedSearchTerm = normalizeTerm( filterValue );
+	const matchSearch = ( string ) => normalizeTerm( string ).indexOf( normalizedSearchTerm ) !== -1;
+	return matchSearch( title ) || keywords.some( matchSearch );
+}
+
+function FillToolbarButton( { name, ...props } ) {
+	return (
+		<Fill name={ `RichText.ToolbarControls.${ name }` }>
+			<ToolbarButton { ...props } />
+		</Fill>
+	);
+}
+
+function FillInserterListItem( props ) {
+	return (
+		<Fill name="Inserter.InlineElements">
+			{ ( { filterValue } ) => {
+				if ( filterValue && ! isResult( props, filterValue ) ) {
+					return null;
+				}
+
+				return <InserterListItem { ...props } />;
+			} }
+		</Fill>
+	);
+}
+
+class Shortcut extends Component {
+	constructor() {
+		super( ...arguments );
+
+		this.onUse = this.onUse.bind( this );
+	}
+
+	onUse() {
+		this.props.onUse();
+		return false;
+	}
+
+	render() {
+		const { character, type } = this.props;
+
+		return (
+			<KeyboardShortcuts
+				bindGlobal
+				shortcuts={ {
+					[ rawShortcut[ type ]( character ) ]: this.onUse,
+				} }
+			/>
+		);
+	}
+}
+
+const FormatEdit = ( { onChange, value } ) => {
+	return (
+		<Fragment>
+			{ getFormatTypes().map( ( { name, edit: Edit, keywords }, i ) => {
+				if ( ! Edit ) {
+					return null;
+				}
+
+				const activeFormat = getActiveFormat( value, name );
+				const isActive = activeFormat !== undefined;
+				const activeAttributes = isActive ? activeFormat.attributes || {} : {};
+
+				return (
+					<Edit
+						key={ i }
+						isActive={ isActive }
+						activeAttributes={ activeAttributes }
+						value={ value }
+						onChange={ onChange }
+						ToolbarButton={ FillToolbarButton }
```

Comment:
I'm still not sure if this is a good idea to pass down both `ToolbarButton` and `InserterListItem` as props. In other places we always import them from one of the packages like `components`, `editor` or `edit-post`.

Reply:
Yeah, maybe that's better, I'm unsure. Trying to think how it would be if `RichText` moves to the `rich-text` packages. You're probably right in that case.